### ISSUE TITLE
Test mod_global_distrib_SUITE:test_in_order_messages_on_multiple_connections_with_bounce fails

### ISSUE THREAD

**Issue by arcusfelis** [esl/MongooseIM#1666]
MongooseIM version: b89252e 
Installed from: source 
Erlang/OTP version: 19.3, travis

```erlang
====== Suite FAILED: mod_global_distrib_SUITE (1 of 23 tests failed)
====== Test name: test_in_order_messages_on_multiple_connections_with_bounce
====== Reason:    {{assertion_failed,assert,is_chat_message,
                       [<<"2">>],
                       {xmlel,<<"message">>,
                           [{<<"from">>,<<"alicE71.944292@localhost/res1">>},
                            {<<"to">>,<<"eve71.944292@localhost/res1">>},
                            {<<"xml:lang">>,<<"en">>},
                            {<<"type">>,<<"chat">>}],
                           [{xmlel,<<"body">>,[],[{xmlcdata,<<"9">>}]}]},
                       "<message from='alicE71.944292@localhost/res1' to='eve71.944292@localhost/res1' xml:lang='en' type='chat'><body>9</body></message>"},
                   [{escalus_new_assert,assert_true,2,
                        [{file,"src/escalus_new_assert.erl"},{line,84}]},
                    {lists,foreach,2,[{file,"lists.erl"},{line,1338}]},
                    {escalus_story,story,4,
                        [{file,"src/escalus_story.erl"},{line,72}]},
                    {test_server,ts_tc,3,
                        [{file,"test_server.erl"},{line,1529}]},
                    {test_server,run_test_case_eval1,6,
                        [{file,"test_server.erl"},{line,1045}]},
                    {test_server,run_test_case_eval,9,
                        [{file,"test_server.erl"},{line,977}]}]}
```

**Comment by arcusfelis** [comment]
One more time

```erlang
====== Test name: test_in_order_messages_on_multiple_connections_with_bounce
====== Reason:    {{assertion_failed,assert,is_chat_message,
                       [<<"2">>],
                       {xmlel,<<"message">>,
                           [{<<"from">>,<<"alicE85.815923@localhost/res1">>},
                            {<<"to">>,<<"eve85.815923@localhost/res1">>},
                            {<<"xml:lang">>,<<"en">>},
                            {<<"type">>,<<"chat">>}],
                           [{xmlel,<<"body">>,[],[{xmlcdata,<<"13">>}]}]},
                       "<message from='alicE85.815923@localhost/res1' to='eve85.815923@localhost/res1' xml:lang='en' type='chat'><body>13</body></message>"},
                   [{escalus_new_assert,assert_true,2,
                        [{file,"src/escalus_new_assert.erl"},{line,84}]},
                    {lists,foreach,2,[{file,"lists.erl"},{line,1338}]},
                    {escalus_story,story,4,
                        [{file,"src/escalus_story.erl"},{line,72}]},
                    {test_server,ts_tc,3,
                        [{file,"test_server.erl"},{line,1529}]},
                    {test_server,run_test_case_eval1,6,
                        [{file,"test_server.erl"},{line,1045}]},
                    {test_server,run_test_case_eval,9,
                        [{file,"test_server.erl"},{line,977}]}]}
```

Link to report
http://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1665/3972/ldap_mnesia.19.3/big/ct_run.test@travis-job-89e4dd1e-32a0-49f8-ad3b-b5e1113bc0fb.2018-01-17_17.37.31/ejabberd_tests.tests.mod_global_distrib_SUITE.logs/run.2018-01-17_17.49.06/mod_global_distrib_suite.test_in_order_messages_on_multiple_connections_with_bounce.html

**Comment by arcusfelis** [comment]
Disabled the test for now #1683

**Comment by arcusfelis** [comment]
should be fixed in #1689
BUT
http://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1708/4087/mysql_redis.19.3/big/ct_run.test@travis-job-869007a9-fe9a-4f62-83ad-2beb6ee02913.2018-02-08_16.50.25/ejabberd_tests.tests.mod_global_distrib_SUITE.logs/run.2018-02-08_17.03.22/mod_global_distrib_suite.test_in_order_messages_on_multiple_connections_with_bounce.html

```erlang
====== Suite FAILED: mod_global_distrib_SUITE (1 of 23 tests failed)
====== Test name: test_in_order_messages_on_multiple_connections_with_bounce
====== Reason:    {{assertion_failed,assert,is_chat_message,
                       [<<"5">>],
                       {xmlel,<<"message">>,
                           [{<<"from">>,<<"alicE43.49749@localhost/res1">>},
                            {<<"to">>,<<"eve43.49749@localhost/res1">>},
                            {<<"xml:lang">>,<<"en">>},
                            {<<"type">>,<<"chat">>}],
                           [{xmlel,<<"body">>,[],[{xmlcdata,<<"8">>}]}]},
                       "<message from='alicE43.49749@localhost/res1' to='eve43.49749@localhost/res1' xml:lang='en' type='chat'><body>8</body></message>"},
                   [{escalus_new_assert,assert_true,2,
                        [{file,"src/escalus_new_assert.erl"},{line,84}]},
                    {lists,foreach,2,[{file,"lists.erl"},{line,1338}]},
                    {escalus_story,story,4,
                        [{file,"src/escalus_story.erl"},{line,72}]},
                    {test_server,ts_tc,3,
                        [{file,"test_server.erl"},{line,1529}]},
                    {test_server,run_test_case_eval1,6,
                        [{file,"test_server.erl"},{line,1045}]},
                    {test_server,run_test_case_eval,9,
                        [{file,"test_server.erl"},{line,977}]}]}
```

@fenek

**Comment by fenek** [comment]
As far as I can see, PR 1708 does not include the fix from 1689.

**Comment by fenek** [comment]
Most probably fixed by #1689. Recently rather only the test case with ungraceful reconnection fails randomly.
